src/scripts/scripts/zone/karazhan/boss_prince_malchezaar.cpp

```diff
@@ -393,7 +393,10 @@ struct boss_malchezaarAI : public ScriptedAI
         }
         else
         {
-            std::vector<InfernalPoint*>::iterator itr = positions.begin()+rand()%positions.size();
+            // Generate new seed (For some reason, rand() was generating same number... casusing infernals to spawn same point)
+            // Infernals spawning together is still possible 1/19 chance of happening
+            srand(time(NULL));
+            std::vector<InfernalPoint*>::iterator itr = positions.begin() + rand() % positions.size();
```

Comment:
Its more randomized but there is still a chance to get two spawns on the same position.